### DISCUSSION TITLE
Question about URL params converted to atoms inside of `Tesla.Env`.

### DISCUSSION THREAD

**Discussion by zorn** [elixir-tesla/tesla#604] [Q&A] [answered]
I'm using Tesla and mocking a chain of network requests for testing pagination with something like:

```
    Tesla.Mock.mock(fn env ->
      if :pageToken in Keyword.keys(env.query) do
        body = #...
        %Tesla.Env{status: 200, body: body}
      else
        body = # ...
        %Tesla.Env{status: 200, body: body}
      end
    end)
```

I was surprised to see that `query` value of `Tesla.Env` was a Keyword list. I would have expected a map with string keys though perhaps it is legal to have the same key twice. 

Anyways, my big concern is that since `Tesla.Env` values will be made during runtime, and the URLs dynamic, could this be an atom memory concern? Is see the use of `to_exsisting_atom` (if I am looking at the right code), but considering the dynamic nature of these URLs and `Tesla.Env` values I'm wondering aloud if this has been a problem for others.

https://github.com/elixir-tesla/tesla/blob/9fbf2215b5945d916ab6f7e757e9623bda5a158b/lib/tesla/middleware/path_params.ex#L37-L41

**Comment by yordis** [comment, marked as answer]
There is an issue about this topic: https://github.com/elixir-tesla/tesla/issues/566  

Ideally, this should be remove and use string
